src/components/NftPreview/NftPreview.tsx

```diff
@@ -34,11 +34,16 @@ const LAYOUT_DIMENSIONS: Record<number, number> = {
 function NftPreview({ nft, collectionId, columns, mobileLayout }: Props) {
   const navigateToUrl = useNavigateToUrl();
 
+  const username = window.location.pathname.split('/')[1];
+  const storage = globalThis?.sessionStorage;
+
   const handleNftClick = useCallback(
     (event: React.MouseEvent<HTMLElement>) => {
-      navigateToUrl(`${window.location.pathname}/${collectionId}/${nft.id}`, event);
+      event.stopPropagation();
+      if (storage) storage.setItem('prevPage', window.location.pathname);
```

Comment:
i know we're trying to avoid using the window due to SSR. @tbezman any recs on the best approach here?